module/userProjects/helper.js

```diff
@@ -2280,6 +2325,369 @@ module.exports = class UserProjectsHelper {
+                        let certificateData = {
+                            recipient : {
+                                id : data.userId,
+                                name : data.userProfile.userName,
+                                type : data.userProfile.userType
+                            },
+                            templateUrl : data.certificate.templateUrl,
+                            issuer : certificateTemplateDetails[0].issuer,
+                            status : UTILS.upperCase(data.certificate.status),
+                            projectId : (data._id).toString(),
+                            projectName : data.title,
+                            programId : (certificateTemplateDetails[0].programId).toString(),
+                            programName : ( data.programInformation && data.programInformation.name ) ? data.programInformation.name : "",
+                            solutionId : (certificateTemplateDetails[0].solutionId).toString(),
+                            solutionName : ( data.solutionInformation && data.solutionInformation.name ) ? data.solutionInformation.name :  "",
+                            completedDate : data.completedDate
+                        };
+                        
+                        const certificateDetails = await certificateService.createCertificate( certificateData );
+                        
+                        if ( !certificateDetails.success || !certificateDetails.data || !certificateDetails.data.ProjectCertificate ) {
+                            throw {
+                                message:  CONSTANTS.apiResponses.CERTIFICATE_GENERATION_FAILED
+                            };
+                        }
+                    
+                        let updateObject = {
+                            "$set" : {}
+                        };
+
+                        //  if transaction id is present.
+                        if ( certificateDetails.data.ProjectCertificate.transactionId &&
+                             certificateDetails.data.ProjectCertificate.transactionId !== "" 
+                            ) {
+                                let transactionIdvalue = certificateDetails.data.ProjectCertificate.transactionId;
+                                const first2 = transactionIdvalue.slice(0, 2);
+                                
+                                if ( first2 === "1-" ) {
+                                    transactionIdvalue = transactionIdvalue.split(/1-(.*)/s)
+                                    updateObject["$set"]["certificate.transactionId"] = transactionIdvalue[1];
+                                } else {
+                                    updateObject["$set"]["certificate.transactionId"] = transactionIdvalue;
+                                }
+                                
+                        }
+                        // update project details certificate details
```

Comment:
@VISHNUDAS-tunerlabs just check if both will come at same time

Reply:
@aks30 tejah confirmed we can get both in response